include/envoy/server/admin.h

```diff
@@ -19,8 +22,45 @@ namespace Server {
  * done in the RouteConfigProviderManagerImpl constructor in source/common/router/rds_impl.cc.
  */
 #define MAKE_ADMIN_HANDLER(X)                                                                      \
-  [this](const std::string& url, Http::HeaderMap& response_headers,                                \
-         Buffer::Instance& data) -> Http::Code { return X(url, response_headers, data); }
+  [this](const std::string& url, Http::HeaderMap& response_headers, Buffer::Instance& data,        \
+         Server::HandlerInfo& handler_info) -> Http::Code {                                        \
+    return X(url, response_headers, data, handler_info);                                           \
+  }
+
+/**
+ * This class is a base class for data which will be sent from admin filter to a handler
+ * in admin impl. Each handler which needs to receive data from admin filter can inherit from
+ * HandlerInfo and build a class which contains the relevant data.
+ */
+class HandlerInfo {
```

Comment:
fixed by commit 61659a39f3beebfd77c81f776386282f85b5c919

Reply:
It looks like as a sink you still need to get the Http::StreamDecoderFilterCallbacks* into your system, but can we get this by plumbing through the AdminFilter* rather than having this empty-base-class/dynamic-cast strategy?  Or @mattklein123 do you have a better idea?